Tools/i18n/pygettext.py

```diff
@@ -341,7 +343,10 @@ def __waiting(self, ttype, tstring, lineno):
             if ttype == tokenize.NAME and tstring in ('class', 'def'):
                 self.__state = self.__suiteseen
                 return
-        if ttype == tokenize.NAME and tstring in opts.keywords:
+        if (
+            ttype == tokenize.NAME and tstring in opts.keywords
+            and (not self.__prev_token or not _is_def_or_class_keyword(self.__prev_token))
+        ):
```

Comment:
The new logic is, if we see one of the gettext keywords __and__ the previous token is not `def` or `class`, only then we transition to `__keywordseen`.